src/crimson/osd/object_context.h

```diff
@@ -203,14 +203,16 @@ class ObjectContext : public Blocker,
   void drop_read() {
     return put_lock_type(RWState::RWREAD);
   }
-  seastar::future<bool> get_recovery_read(bool can_wait = false) {
-    if (bool acquired = rwstate.get_recovery_read();
-        acquired || !can_wait) {
-      return seastar::make_ready_future<bool>(acquired);
+  bool get_recovery_read() {
+    return rwstate.get_recovery_read();
+  }
+  seastar::future<> wait_recovery_read() {
```

Comment:
This seems to have been modified by commit "crimson: use tri_mutex for guarding ObjectContext", maybe it should be merged with that commit:)

Reply:
but this change is orthogonal to the tri_mutex change. i think they should be different commits.